src/pages/[platform]/build-a-backend/add-aws-services/logging/set-up-logging/index.mdx

```diff
@@ -104,29 +104,20 @@ You will need to create a log group in Amazon CloudWatch to send logs to. You ca
 
 Below is a sample CDK construct to create the Amazon CloudWatch log group as well as creating and assigning the permission policies to Amplify roles.
 
-The `<log-group-name>` and `<region>` configured in the CDK construct will be used later to initialize the Amplify Logger plugin.
+Create the `RemoteLoggingConstraintsConstruct` custom CDK construct as the following:
 
-### Replace the placeholder values with your own values:
-
-- `<log-group-name>` is the log group that logs will be sent to. Note that this CDK construct sample includes logic to create the CloudWatch log group that you may have already created in previous steps.
-- `<amplify-authenticated-role-name>` and `<amplify-unauthenticated-role-name>` are Amplify roles created as part of Amplify Auth configuration via Amplify CLI.
-
-```ts
+```ts title="amplify/custom/RemoteLoggingConstraintsConstruct/resource.ts"
 import * as cdk from "aws-cdk-lib"
 import { Construct } from "constructs"
 import * as logs from "aws-cdk-lib/aws-logs"
-import * as path from "path"
 import * as iam from "aws-cdk-lib/aws-iam"
 
 export class RemoteLoggingConstraintsConstruct extends Construct {
-  constructor(scope: Construct, id: string, props: RemoteLoggingConstraintProps) {
+  constructor(scope: Construct, id: string, authRoleName: string, unAuthRoleName: string, logGroupName: string) {
```

Comment:
CDK's pattern is to use 3 args: scope, id, and props. are we missing the props type?

Reply:
I had initially copied over the code which had `props: RemoteLoggingConstraintProps` and ran into type definition issues on `RemoteLoggingConstraintProps`. I believe we are missing the props type. Was it supposed to be defined in this file?

If so, I can define **RemoteLoggingConstraintProps** and add `authRoleName` and `unAuthRoleName` within this type.